contracts/tokenfactory-issuer/src/contract.rs

```diff
@@ -39,7 +40,19 @@ pub fn instantiate(
     IS_FROZEN.save(deps.storage, &false)?;
 
     match msg {
-        InstantiateMsg::NewToken { subdenom } => {
+        InstantiateMsg::NewToken { subdenom, metadata } => {
```

Comment:
Do we need validation to make sure that the base_denom is actually in the metadata?  Or will that be done already by the tokenfactory module?